File: SchoolWebApp-API/Project.Core/Entities/Settings/GlobalSetting.cs
using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations;

namespace SchoolWebApp.Core.Entities.Settings
{
    public class GlobalSetting : Base
    {
        [Required]
        [StringLength(100)]
        public required string Module { get; set; }

        [Required]
        [StringLength(100)]
        public required string SettingKey { get; set; }

        [Required]
        [StringLength(500)]
        public required string SettingValue { get; set; }

        [StringLength(255)]
        public string? Description { get; set; }
    }
}

File: SchoolWebApp-Client/src/app/settings/models/global-setting.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class GlobalSetting extends ResourceModel<GlobalSetting> {
    public module?: string;
    public settingKey?: string;
    public settingValue?: string;
    public description?: string;

    constructor(model?: Partial<GlobalSetting>) {
        super(model);
    }
}
